### PR TITLE
[Response to Reviewers] Update dimensionality reduction figures

### PR DESCRIPTION
**Pull request by gwaybio**
This PR accomplishes two, interrelated items, that I perform in response to two separate reviewer comments.

### Reviewer comment

> Fig 2A – From this figure it looks like “elongated”, “large” and “interphase” (rather than “metaphase”) appear to be most distinct for CellProfiler features; however, looking at the individual maps in Fig S2, the split of “interphase” is easier to see. I feel it would be worth replacing Fig 2A with S2A as these show the distributions more clearly.

In response, I moved S2A to Fig 2A, and I updated Figure S2

#### Updated Figure 2

main_figure_2_umap_and_correlation

#### Updated Supplementary UMAP figure

supplementary_umap

### Reviewer comment

> > The poor single-cell predictions feels inevitable based on the limited phenotype distinctions in Fig 2A. Were alternatives to UMAP, such as tSNE, tested to see if they yielded more clear distinctions?

In response, I fit and trained several t-SNE models (with different perplexities). Here is one example with `perplexity=40`

tsne_figure_perplexity_40

The results are interesting, and we will decide how to proceed in a future conversation

### CODE REVIEW

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import sys
+
+sys.path.append("../utils")
+from split_utils import get_features_data
+
+
+# In[2]:
+
+
+np.random.seed(1234)
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/apply_tsne.py:26]: Why not random seed `0`? I have found that through previous PR reviews that Way Lab convention is to use `0` as random seed specificaly.

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+# In[3]:
+
+
+output_file = pathlib.Path("evaluations", "tsne_embeddings.csv.gz")
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/apply_tsne.py]: Is there a specific reason to use compressed CSV over a parquet file?

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+# ## Apply tSNE
+# 
+# We test different perplexities ranging from 2 to 300.
+# 
+# From scikit-learn (https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html):
+# 
+# > The perplexity is related to the number of nearest neighbors that is used in other manifold learning algorithms. Larger datasets usually require a larger perplexity. Consider selecting a value between 5 and 50. Different values can result in significantly different results. The perplexity must be less than the number of samples.
+# 
+# We do not know what the appropriate value of perplexity is for our dataset, so we will test several
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/apply_tsne.py]:
```suggestion
# We do not know what the appropriate value of perplexity is for our dataset, so we will test several.
```

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+# In[6]:
+
+
+tsne_embedding_df = []
+list_of_perplexities = [2, 10, 15, 30, 40, 60, 80, 100, 150, 300]
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/apply_tsne.py:89]: Why did you choose these integers specifically? I could see using list comprehension to make a much larger list. I recommend including a code comment explaining why you have a list of these integers.

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+print(tsne_embedding_df.shape)
+tsne_embedding_df.head()
+
+
+# In[7]:
+
+
+# Output file
+tsne_embedding_df.to_csv(output_file, sep="\t", index=False)
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/apply_tsne.py:134]: Your output file extension above has a `csv.gz` extension but in this code you have `sep="\t"` which would mean you are saving a `tsv.gz`. Recommend updating the extension in `output_file` to reflect this.

File: 7.figures/nbconverted/SupplementaryFigure_tsne_figures.r [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+)
+
+focus_corr_colors = c(
+    "TRUE" = "blue",
+    "FALSE" = "orange"
+)
+focus_corr_labels  = c(
+    "TRUE" = "Yes",
+    "FALSE" = "No"
+)
```
Comment (jenna-tomkinson) [inline 7.figures/nbconverted/SupplementaryFigure_tsne_figures.r]: Do you need this since these figures don't have correlation plots?

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! I added a few comments for you to address.

I think before we merge, should we not pick a perplexity we want to include in the manuscript? Maybe it is okay to have them all, but from my perspective, they all look pretty similar (FYI some plots could not render in GitHub). 

I think these results are similar to the UMAP where CellProfiler features look to have more heterogeneity compared to DP and CP_CP. This will be a really nice supplemental figure!

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py (hunk 1 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/apply_tsne.py:26]: we do not have a convention on random seeds - for whatever reason, i always use 1234

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py (hunk 2 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/apply_tsne.py]: habit! IMO it's ok to keep compressed csv

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py (hunk 3 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/apply_tsne.py]: will address! Thanks for the careful eye 👁️

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py (hunk 4 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/apply_tsne.py:89]: I will add a comment!

I chose to span a range of values. I tested a bunch by eye - I wanted to see the clustering go from not working to working to not working

File: 3.evaluate_model/scripts/nbconverted/apply_tsne.py (hunk 5 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/apply_tsne.py:134]: this is a great catch! I will fix

File: 7.figures/nbconverted/SupplementaryFigure_tsne_figures.r (hunk 6 above)
Comment (gwaybio) [reply 7.figures/nbconverted/SupplementaryFigure_tsne_figures.r]: yep! Good catch

**Comment by gwaybio** [comment]
> I think before we merge, should we not pick a perplexity we want to include in the manuscript?

I agree - we will not include all. You will find in the response to reviewers document that I have selected 40.

Thanks for the review! I am merging now
